File: src/al5d_sub.cpp
//#include <cstdio>
#include "RobotArm.h"
#include "ros/ros.h"
#include "std_msgs/Float32MultiArray.h"


RobotArm arm;


void al5d_callback(const std_msgs::Float32MultiArray::ConstPtr& msg) {

    arm.commandRobotUsingIK(
        msg->data[0],
        msg->data[1],
        msg->data[2],
        msg->data[3],
        msg->data[4],
        msg->data[5],
        msg->data[6]);
    
}

int main(int argc, char **argv) {
    
    arm.create();

    ros::init(argc, argv, "al5d_sub");

    ros::NodeHandle n;

    ros::Subscriber sub = n.subscribe("al5d", 10, al5d_callback);

    ros::spin();

    return 0;
    
}

File: scripts/al5d_sub.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import Float32MultiArray

import arlo.output.al5d as al5d



arm = al5d.RobotArm()


def al5d_callback(msg):
    data = msg.data
    
    #arm.move_IK(data[0],data[1],data[2],data[3],data[4],data[5])
    arm.displace(data)

def main():

    arm.create()
    
    rospy.init_node('al5d_sub', anonymous=True)
    
    rospy.Subscriber('al5d', Float32MultiArray, al5d_callback) 
    
    rospy.spin()

    arm.destroy()



if __name__ == '__main__':
    try:
        main()
    except rospy.ROSInterruptException:
        pass
